### PR TITLE
Fix remove_dxes_except corner cases

### PR DESCRIPTION
**Pull request by JulienVdG**
The parseBlackList function create a regex with several string option,
FindFilePredicate encapsulate it to match whole word only but we have
to take care of the blacklist case where the regex string include '|'.

Signed-off-by: Julien Viard de Galbert <julien.viard-de-galbert@itrenew.com>

### CODE REVIEW

File: pkg/visitors/remove_test.go [hunk 1]
```diff
@@ -112,3 +112,54 @@ func TestRemoveExcept(t *testing.T) {
...
 
 }
+
+func TestRemoveExceptBlackList(t *testing.T) {
+	// The following blacklist contains corner case:
+	// - start, mid or end of existing names but no full name
+	// - start, mid or end of existing GUID but no full GUID
+	// so it should behave as an empty blacklist, ie remove all
+	var blacklists = []string{
+		`INEXISTING_FILENAME
```
Comment (GanShun) [inline pkg/visitors/remove_test.go]: Could you use "" with \n here? or does that not work? That would make the formatting a little nicer :)

**Comment by GanShun** [comment]
This looks good! just one nit

File: pkg/visitors/remove_test.go (hunk 1 above)
Comment (JulienVdG) [reply pkg/visitors/remove_test.go]: Right, at first I had a long list and no array... fixed

**Comment by JulienVdG** [comment]
:angry: I clicked rebase and merge ... so why did github merge the commit instead of rebase it !!!

**Pull request merged**